reports/push_notifications/push_user_er.sql

```diff
@@ -0,0 +1,63 @@
+WITH sample AS (
+   SELECT client_id,
+          date_parse(submission_date_s3, '%Y%m%d') as normalized_submission_date,
+          popup_notification_stats['web-notifications'].action_1 AS push_acceptance,
+          web_notification_shown
+   FROM main_summary
+   WHERE submission_date_s3 > date_format(CURRENT_DATE - interval '6' MONTH, '%Y%m%d')
+     AND submission_date_s3 < date_format(CURRENT_DATE, '%Y%m%d')
+     AND sample_id='42'
+),
+push_accepters AS (
+   SELECT DISTINCT client_id,
+                   normalized_submission_date AS eligibility_start,
+                   normalized_submission_date + interval '30' DAY AS eligibility_end
+   FROM sample
+   WHERE push_acceptance > 0
+),
+combined_push_users AS (
+   SELECT DISTINCT p.client_id,
+                   eligibility_start,
+                   eligibility_end
+   FROM sample s
+   INNER JOIN push_accepters p ON s.client_id=p.client_id
+   WHERE normalized_submission_date BETWEEN eligibility_start AND eligibility_end
+   AND web_notification_shown > 0
+),
+all_dau AS (
+   SELECT DISTINCT client_id,
+                   normalized_submission_date AS activity_date
+   FROM sample
+),
+push_dau AS (
+  -- Presto can't do RANGE window functions nor can it do count distinct in a window, so the array_agg here and in push_mau is a workaround hack.
```

Comment:
Have you thought about using [HLL](https://github.com/vitillo/presto-hyperloglog/)?